compiler-k2/src/test/kotlin/com/squareup/anvil/compiler/k2/fir/constructor/inject/FirInjectConstructorFactoryGenerationExtensionTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.squareup.anvil.compiler.k2.fir.constructor.inject
+
+import com.squareup.anvil.compiler.testing.CompilationMode
+import com.squareup.anvil.compiler.testing.CompilationModeTest
+import dagger.internal.Provider
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.TestFactory
+
+class FirInjectConstructorFactoryGenerationExtensionTest : CompilationModeTest(
+  CompilationMode.K2(useKapt = false),
+) {
+  @TestFactory
+  fun `factory class is generated for @Inject annotation`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test
+      class InjectClass @javax.inject.Inject constructor()
+      """.trimIndent(),
+    ) {
+      classGraph.allClassesAsMap["com.squareup.test.InjectClass_Factory"].shouldNotBeNull()
+    }
+  }
+
+  @TestFactory
+  fun `factory class is not generated if class is not annotated`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test
+      class InjectClass()
+      """.trimIndent(),
+    ) {
+      classGraph.allClassesAsMap["com.squareup.test.InjectClass_Factory"].shouldBeNull()
+    }
+  }
+
+  @TestFactory
+  fun `factory class creates source class with values from provider`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test
+      class InjectClass @javax.inject.Inject constructor(
+        val param0: String,
+        val param1: Int,
+      )
+      """.trimIndent(),
+    ) {
+      val expectedString = "ExpectedString"
+      val expectedInt = 77
+      val injectFactoryClass =
+        classGraph.allClassesAsMap["com.squareup.test.InjectClass_Factory"]!!.loadClass()
```

Comment:
nit: You can just do `classLoader.loadClass(...)` for these.

`injectClassFactory` and `injectClass` are actually defined in `CommonNames` and give you the `Class<*>` instances.  I just forgot to make them public.